src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/CVTextMouseActionData.cs

```diff
@@ -0,0 +1,544 @@
+        public TextCaseRule textCaseRule = TextCaseRule.Matches;
+
+        /**
+         * optionally limit the rect area where the TextData can be detected
+         */
+        [CanBeNull]
+        public CVWithinRect withinRect;
+
+        /**
+         * the list of actions to perform at this location
+         * Note: This does NOT re-evaluate the position of the CVText on each action.
+         * This is modeled as a list to allow you to click and/or un-click buttons on this Text location without re-evaluating the CV text match in between if you don't want to.
+         * If you want to validate a mouse visual effect in your criteria, and then mouse up in the next bot segment; that is also a normal way to use this system with just 1 list action per segment.
+         *
+         * If you want to perform click and drag mouse movements, you cannot and should not try to do that with this list.  You need to create 2 separate bot segments.
+         * Mouse down in one bot segment, then mouse up as a separate bot segment.
+         */
+        public List<CVMouseActionDetails> actions;
+
+        private bool _isStopped;
+
+        private float _nextActionTime = float.MinValue;
+        private int _nextActionIndex = 0;
+
+        private bool _resultReceived = false;
+        private RectInt? _cvResultsBoundsRect = null;
+
+        // if an entry is missing, no request for that segment in progress
+        private bool _requestInProgress = false;
+        private Action _requestAbortAction = null;
+        private readonly object _syncLock = new();
+
+        private string _lastError = null;
+
+        public bool IsCompleted()
+        {
+            return _isStopped;
+        }
+
+        public void ReplayReset()
+        {
+            _isStopped = false;
+            _nextActionTime = float.MinValue;
+            _nextActionIndex = 0;
+            _resultReceived = false;
+            _cvResultsBoundsRect = null;
+            lock (_syncLock)
+            {
+                if (_requestAbortAction != null)
+                {
+                    _requestAbortAction.Invoke();
+                    _requestAbortAction = null;
+                }
+
+                _requestInProgress = false;
+            }
+        }
+
+        public void StartAction(int segmentNumber, Dictionary<long, ObjectStatus> currentTransforms, Dictionary<long, ObjectStatus> currentEntities)
+        {
+            // get the CV evaluate started...
+            RequestCVTextEvaluation(segmentNumber);
+        }
+
+        [Serializable]
+        private struct ResultTreeNode
+        {
+            public List<ResultTreeNode> children;
+
+            public List<int> indexGraph;
+
+            public RectInt? boundsRect;
+        }
+
+        private void RequestCVTextEvaluation(int segmentNumber)
+        {
+            lock (_syncLock)
+            {
+                if (!_requestInProgress)
```

Comment:
I agree with Aaron that the amount of indentation is making it slower for me to read. 

Would it be possible to have this bit be an early exit for eample?

if(_requestInProgress)
  return;